### PR TITLE
Split engine test suite to improve IDE performance

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: ktor-server/ktor-server-jetty/jvm/test/io/ktor/tests/server/jetty/JettyServletApplicationEngine.kt [hunk 1]
```diff
@@ -7,42 +7,24 @@ package io.ktor.tests.server.jetty
...
-    }
-}
-
-// the factory and engine are only suitable for testing
-// you shouldn't use it for production code
-
-private class Servlet(private val async: Boolean) :
+/**
+ * The factory and engine are only suitable for testing. You shouldn't use it for production code.
+ */
```
Comment (Ololoshechkin) [inline ktor-server/ktor-server-jetty/jvm/test/io/ktor/tests/server/jetty/JettyServletApplicationEngine.kt:15]: Is there a way to use it if it's internal in `*.test.*` package?

Comment (Ololoshechkin) [reply ktor-server/ktor-server-jetty/jvm/test/io/ktor/tests/server/jetty/JettyServletApplicationEngine.kt:15]: Upd: Well, I get the point. I think we should make some annotation for that, like `@TestOnly` and check that it's being used only in test module

File: ktor-server/ktor-server-test-host/jvm/src/io/ktor/server/testing/suites/CompressionTestSuite.kt [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+import java.io.*
+import java.util.zip.*
+import kotlin.test.*
+
+abstract class CompressionTestSuite<TEngine : ApplicationEngine, TConfiguration : ApplicationEngine.Configuration>(
+    hostFactory: ApplicationEngineFactory<TEngine, TConfiguration>
+) : EngineTestBase<TEngine, TConfiguration>(hostFactory) {
+
+    @Test
+    fun testLocalFileContentWithCompression() {
```
Comment (Ololoshechkin) [inline ktor-server/ktor-server-test-host/jvm/src/io/ktor/server/testing/suites/CompressionTestSuite.kt:29]: Nit: I was thinking of how to make this similar-looking tests table-driven or at least extract some common code, but maybe it's not worth it as there are quite a few parameters to introduce.
Maybe let's leave it as it is.

File: ktor-server/ktor-server-netty/jvm/test/io/ktor/tests/server/netty/NettyEngineTest.kt [hunk 3]
```diff
@@ -5,12 +5,45 @@
...
+}
+
+class NettyContentTest : ContentTestSuite<NettyApplicationEngine, NettyApplicationEngine.Configuration>(Netty) {
+    init {
+        enableSsl = true
+    }
+
+    override fun configure(configuration: NettyApplicationEngine.Configuration) {
+        configuration.shareWorkGroup = true
+    }
```
Comment (Ololoshechkin) [inline ktor-server/ktor-server-netty/jvm/test/io/ktor/tests/server/netty/NettyEngineTest.kt:27]: Nit: Maybe extract a a parent class with enableSsl = true, configuration.shareWorkGroup = true ?

File: ktor-server/ktor-server-test-host/jvm/src/io/ktor/server/testing/suites/CompressionTestSuite.kt (hunk 2 above)
Comment (e5l) [reply ktor-server/ktor-server-test-host/jvm/src/io/ktor/server/testing/suites/CompressionTestSuite.kt:29]: I'm experimenting with that on the client-side, and haven't found any good solutions yet.

Maybe we should look in different test frameworks, something like `spec`. But I'm not sure if they work with multiplatform.

**Pull request merged**